src/components/FunctionsPage/Functions.js

```diff
@@ -248,10 +250,19 @@ const Functions = ({
   }
 
   const handleDeployFunctionSuccess = () => {
+    const { name, tag } = functionsStore.newFunction.metadata
+
     setFunctionsPanelIsOpen(false)
     removeNewFunction()
 
-    return refreshFunctions().then(() => {
+    return refreshFunctions().then(functions => {
+      const [currentItem] = functions.filter(
+        func => func.name === name && func.tag === tag
+      )
```

Comment:
If you use `Array#find` instead of `Array#filter` you will not need to destructure an array.
```suggestion
      const currentItem = functions.find(
        func => func.name === name && func.tag === tag
      )
```